presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/EncoderFieldValueProvider.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka.encoder;
+
+import io.airlift.slice.Slice;
+import io.prestosql.spi.block.Block;
+
+import java.nio.ByteBuffer;
+
+public interface EncoderFieldValueProvider
+{
+    EncoderColumnHandle getColumn();
+
+    boolean getBoolean();
+
+    long getLong();
+
+    int getInt();
+
+    short getShort();
+
+    byte getByte();
+
+    double getDouble();
+
+    float getReal();
+
+    String getVarchar();
+
+    String getChar();
+
+    ByteBuffer getVarbinary();
+
+    Slice getSlice();
+
+    Block getBlock();
```

Comment:
these (and `getChar`) are unused

also, if the calling code could handle presto native types (slice, block), this class would be unnecessary

Reply:
right, this class really doesn't do a whole lot I guess